rasa_nlu/utils/md_to_json.py

```diff
@@ -32,63 +42,71 @@ class MarkdownToJson(object):
     def __init__(self, file_name):
         self.file_name = file_name
         # set when parsing examples from a given intent
-        self.current_intent = None
+        self.current_title = None
+        self.current_section = None
         self.common_examples = []
-        self.entity_synonyms = []
-        self.load()
+        self.entity_synonyms = defaultdict(set)
+        self.regex_patterns = []
+
+    def find_section_header(self, line):
+        """Checks if the current line contains a section header and returns the section and the title."""
+        for name, regex in section_regexes.items():
+            match = re.search(regex, line)
+            if match is not None:
+                return name, match.group(1)
+        return None
+
+    def make_json(self):
```

Comment:
This shouldn't be done in here, as it shouldn't concern the markdown reader (for that matter the class should be renamed as well). The classes only concern should be reading md and creating an instance of `TrainingData`. Something like `TrainingData(trainings_examples, synonyms, regex_features).as_json()` should be used to convert to json.